plugins/CorePluginsAdmin/CorePluginsAdmin.php

```diff
@@ -28,17 +28,16 @@ public function registerEvents()
             'AssetManager.getStylesheetFiles'        => 'getStylesheetFiles',
             'System.addSystemSummaryItems'           => 'addSystemSummaryItems',
             'Translate.getClientSideTranslationKeys' => 'getClientSideTranslationKeys',
-            'PluginManager.pluginActivated'          => 'onPluginActivated',
-            'PluginManager.pluginInstalled'          => 'addPluginChanges',
             'Updater.componentUpdated'               => 'addPluginChanges',
-            'PluginManager.pluginUninstalled'        => 'removePluginChanges'
```

Comment:
Is it correct here to remove the hook for the plugin uninstallation?

Reply:
@michalkleiner Yes that's intentional, plugins are [deactivated before they are uninstalled](https://github.com/matomo-org/matomo/blob/f4209ffb53ad7949e1efd50c159223d17371b927/core/Plugin/Manager.php#L621) so the `removePluginChanges()` method will still be run when uninstalling, but also now if a plugin is deactivated but not uninstalled.

Reply:
This is actually not fully correct. Uninstalling a plugin triggers the `deactivate` method within the plugin class (if found). But that will actually not trigger the event.
The UI only provides an uninstall link for deactivated plugins, but there might be a scenario where you can uninstall a plugin in the UI without deactivating it first. Haven't tested that, but when you remove the plugin directly in the filesystem, I think the system will report it as broken, with the option to uninstall it. But the deactivate might not be triggered in that case.
Also there could be other scanarios where an activated plugin is uninstalled without a manual deactivation before. So might be better to keep that event as well.

Reply:
@bx80 do we want to return the event hook? It should cause no harm, right? Or would it effectively run twice and cause problems?